homeassistant/helpers/script.py

```diff
@@ -474,37 +537,71 @@ async def _async_call_service_step(self):
             await service_task
             return
 
-        async def async_cancel_service_task():
-            # Stop service task and wait for it to finish.
-            service_task.cancel()
-            try:
-                await service_task
-            except Exception:  # pylint: disable=broad-except
-                pass
-
-        # No call limit so watch for a stop request.
-        stop_task = self._hass.async_create_task(self._stop.wait())
-        try:
-            await asyncio.wait(
-                {service_task, stop_task}, return_when=asyncio.FIRST_COMPLETED
+        await self._async_run_long_action(service_task)
+
+    async def _async_repeat_step(self):
+        """Repeat a sequence."""
+
+        description = self._action.get(CONF_ALIAS, "sequence")
+        repeat = self._action[CONF_REPEAT]
+
+        async def async_run_sequence(iteration, extra_msg="", extra_vars=None):
+            self._log("Repeating %s: Iteration %i%s", description, iteration, extra_msg)
+            repeat_vars = {"repeat": {"first": iteration == 1, "index": iteration}}
+            if extra_vars:
+                repeat_vars["repeat"].update(extra_vars)
+            task = self._hass.async_create_task(
+                # pylint: disable=protected-access
+                self._script._repeat_script[self._step].async_run(
+                    # Add repeat to variables. Override if it already exists in case of
+                    # nested calls.
+                    {**(self._variables or {}), **repeat_vars},
+                    self._context,
+                )
             )
-        # If our task is cancelled, then cancel service task, too. Note that if service
-        # task is cancelled otherwise the CancelledError exception will not be raised to
-        # here due to the call to asyncio.wait(). Rather we'll check for that below.
-        except asyncio.CancelledError:
-            await async_cancel_service_task()
-            raise
-        finally:
-            stop_task.cancel()
+            await self._async_run_long_action(task)
+
+        if CONF_COUNT in repeat:
+            count = repeat[CONF_COUNT]
+            if isinstance(count, template.Template):
+                try:
+                    count = vol.Coerce(int)(count.async_render(self._variables))
```

Comment:
We can just use `int(…)` and catch `ValueError`
```suggestion
                    count = int(count.async_render(self._variables))
```

Reply:
True. Will do.